python/main.py

```
@@ -18,14 +22,40 @@
     allow_headers=["*"],
 )
 
+def save_items(name, category, image_filename):
+    with open('items.json', 'r') as f:
+        data = json.load(f)
+    with open("items.json", "w") as f:
+        data["items"].append({"name": name, "category": category, "image_filename": image_filename})
+        json.dump(data, f)
+
 @app.get("/")
 def root():
     return {"message": "Hello, world!"}
 
 @app.post("/items")
-def add_item(name: str = Form(...)):
+def add_item(name: str = Form(...), category: str = Form(...), image: UploadFile = File(...)):
     logger.info(f"Receive item: {name}")
-    return {"message": f"item received: {name}"}
+    logger.info(f"Receive item: {category}")
+    logger.info(f"Receive item: {image}")
+    image = images / image.filename
+    with open(image, "rb") as im:
+        image_hash = hashlib.sha256(im.read()).hexdigest()
+    image_filename = str(image_hash) + ".jpg"
+    save_items(name, category, image_filename)
+    return {"message": f"item received: {name}, {category}, {image}"}
+
+@app.get("/items")
+def show_item():
+    with open("items.json", "r") as f:
+        data = json.load(f)
+    return data
+
+@app.get("/items/{item_id}")
+def item_id(item_id: int):
+    with open("items.json", "r") as f:
+        data = json.load(f)
+    return data["items"][item_id-1]
```

We should check if the item index is valid, otherwise we should return an error.